warehouse/oidc/tasks.py

```diff
@@ -65,3 +68,21 @@ def compute_oidc_metrics(request):
             .count(),
             tags=[f"publisher:{discriminator}"],
         )
+
+
+@tasks.task(ignore_result=True, acks_late=True)
+def delete_expired_oidc_macaroons(request):
+    # Purge all API tokens minted using OIDC Trusted Publishing with a creation time
+    # more than 15 minutes ago. Since OIDC-minted macaroons expire 15 minutes after
+    # creation, this task cleans up all the expired tokens that have accumulated since
+    # the last time this task was run.
+    (
+        request.db.query(Macaroon)
+        .filter(Macaroon.oidc_publisher_id.isnot(None))
+        .filter(
+            # The expiration time (created + 15 mins) is in the past
+            Macaroon.created + timedelta(minutes=15)
+            < datetime.now(tz=timezone.utc)
+        )
+        .delete(synchronize_session=False)
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** capture the result of the delete operation and emit as a metric from [`result.rowcount`](https://docs.sqlalchemy.org/en/20/tutorial/data_update.html#getting-affected-row-count-from-update-delete) so we have an idea of how many are being removed each run.‏

Reply:
Added